app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -0,0 +1,217 @@
+import SmartTransactionsController from '@metamask/smart-transactions-controller';
+import {
+  Fee,
+  SmartTransactionStatuses,
+  SmartTransaction,
+} from '@metamask/smart-transactions-controller/dist/types';
+import type { Hex } from '@metamask/utils';
+import {
+  TransactionController,
+  TransactionMeta,
+  TransactionParams,
+} from '@metamask/transaction-controller';
+import log from 'loglevel';
+
+import { decimalToHex } from '../../../../shared/modules/conversion.utils';
+import {
+  SMART_TRANSACTION_CONFIRMATION_TYPES,
+  ORIGIN_METAMASK,
+} from '../../../../shared/constants/app';
+
+export type SubmitSmartTransactionRequest = {
+  transactionMeta: TransactionMeta;
+  smartTransactionsController: SmartTransactionsController;
+  transactionController: TransactionController;
+  isSmartTransaction: boolean;
+  controllerMessenger: any;
+  featureFlags: Record<string, any>;
+};
+
+export async function submitSmartTransactionHook(
+  request: SubmitSmartTransactionRequest,
+) {
+  const {
+    transactionMeta,
+    smartTransactionsController,
+    transactionController,
+    isSmartTransaction,
+    controllerMessenger,
+    featureFlags,
+  } = request;
+  log.info('Smart Transaction - Executing publish hook', transactionMeta);
+  const isDapp = transactionMeta?.origin !== ORIGIN_METAMASK;
```

Comment:
Is there a case where `transactionMeta` may be undefined?

Reply:
Good point, I've changed it to: `const isDapp = transactionMeta.origin !== ORIGIN_METAMASK;`